aten/src/ATen/test/cuda_allocator_test.cpp

```diff
@@ -5,6 +5,84 @@
 
 #include <ATen/test/allocator_clone_test.h>
 
+#include <torch/csrc/cuda/CUDAPluggableAllocator.h>
+
+std::unordered_map<void*, ssize_t> allocation_sizes;
+
+void* logging_malloc(ssize_t size, int device, cudaStream_t stream) {
```

Comment:
logging malloc/free option could also be very useful even not as part of test! (and ptr can be used in place of unique allocation tag)